3.5/alpine/Dockerfile

```diff
@@ -40,12 +42,16 @@ RUN apk add --no-cache --virtual .build-deps  \
 		\( -type d -a -name test -o -name tests \) \
 		-o \( -type f -a -name '*.pyc' -o -name '*.pyo' \) \
 		-exec rm -rf '{}' + \
-	&& rundeps=$(scanelf -R -n --nobanner /usr/local/  \
-		| awk '{gsub(/,/,"\n",$2); print $2}' \
-		| sort -u | sed 's/^/so:/' | while read dep; do \
-			apk info --installed -q $dep && echo $dep; \
-		done | sort -u) \
-	&& apk add --virtual .python-rundeps $rundeps \
+	&& runDeps="$( \
+		scanelf --needed --nobanner --recursive /usr/local \
+			| awk '{ gsub(/,/, "\n", $2); print "so:" $2 }' \
+			| sort -u \
+			| while read dep; do \
+				apk info --installed -q "$dep" && echo "$dep"; \
+			done \
+			| sort -u \
```

Comment:
Is this second `sort -u` actually necessary?  Our `while` loop has just filtered the output of the first `sort`, so this one should be a no-op.